core-bundle/src/Resources/contao/widgets/RootPageDependentSelect.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao;
+
+class RootPageDependentSelect extends SelectMenu
+{
+	public function generate(): string
+	{
+		$fields = array();
+		$originalLabel = $this->strLabel;
+		$cssClasses = 'tl_select tl_chosen';
+		$rootPages = PageModel::findByPid(0, array('order' => 'sorting'));
+		$wizard = StringUtil::deserialize($this->wizard);
+
+		$this->blankOptionLabel = $GLOBALS['TL_LANG']['tl_module'][sprintf('%sBlankOptionLabel', $this->name)];
+
+		foreach ($rootPages as $rootPage)
+		{
+			$label = sprintf('%s (%s)', $rootPage->title, $rootPage->language);
+			$this->arrOptions[0]['label'] = sprintf($this->blankOptionLabel, $label);
+			$this->strLabel = $label;
+
+			$fields[] = sprintf(
+				'<select name="%s[]" id="ctrl_%s" class="%s%s"%s onfocus="Backend.getScrollOffset()">%s</select>%s',
+				$this->strName,
+				sprintf('%s-%s', $this->strId, $label),
+				$cssClasses,
+				($this->strClass ? ' ' . $this->strClass : ''),
+				$this->getAttributes(),
+				implode('', $this->getOptions($rootPage->id)),
+				$wizard[$rootPage->id] ?? ''
+			);
+		}
+
+		$this->strLabel = $originalLabel;
+
+		return implode('', $fields);
+	}
+
+	protected function isSelected($option): string
+	{
+		if (empty($this->varValue) && empty($_POST) && ($option['default'] ?? null))
+		{
+			return static::optionSelected((string) 1, 1);
```

Comment:
You need this to generate the string ` selected`, right? This should IMHO get rid of the implicit casts, then:

```suggestion
			return static::optionSelected('1', '1'); // selected
```

Or just use the value:

```suggestion
			return ' selected';
```

Reply:
Adjusted in a4762f93b.